FormalConjectures/Paper/CasasAlvero.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Casas-Alvero Conjecture
+
+*Reference:*
+* [The Casas-Alvero conjecture for infinitely many degrees](https://arxiv.org/pdf/math/0605090)
+* [MathOverflow](https://mathoverflow.net/questions/27851)
+
+The Casas-Alvero conjecture states that if a univariate polynomial `P` of degree `d` over a field
+of characteristic zero shares a root with each of its derivatives up to order `d-1`, then `P` must
+be of the form `(X - α)^d` for some `α`.
+
+The conjecture has been proven for:
+* Degrees `d ≤ 8`
+* Degrees of the form `p^k` where `p` is prime
+* Degrees of the form `2p^k` where `p` is prime
+
+The conjecture is false in positive characteristic `p` for polynomials of degree `d > p`.
+
+The conjecture is now claimed to be proven in this paper:
+* [Proof of the Casas-Alvero conjecture: Soham Ghosh)](https://arxiv.org/pdf/2501.09272)
+
+-/
+
+open Polynomial
+
+/--
+A polynomial `P` satisfies the Casas-Alvero property if it shares a root with each
+of its Hasse derivatives up to order `d-1`, where `d` is the degree of `P`.
+-/
+def HasCasasAlveroProp {K : Type*} [Field K] (P : K[X]): Prop :=
+  ∀ i ∈ Finset.range P.natDegree, ∃ α : K, IsRoot P α ∧ IsRoot (hasseDeriv i P) α
+
+/--
+The Casas-Alvero property is equivalent to the polynomial not being coprime with any of its
+Hasse derivatives up to order `d-1`.
+-/
+@[category API, AMS 12]
+lemma casas_alvero_iff_coprime {K : Type*} [Field K] (P : K[X]) (hP : Monic P) :
+  HasCasasAlveroProp P ↔ ∀ i ∈ Finset.range P.natDegree, ¬ IsCoprime P (hasseDeriv i P) := by
+  sorry
```

Comment:
I think we need to assume K is CharZero and quantify over all fields; otherwise we might only prove this after we've proven the conjecture. Please check out my proof [here](https://github.com/shamEiNew/formal-conjectures/compare/CasasAlveroConjecture...alreadydone:formal-conjectures:patch-2).